### PR TITLE
Add transaction scaffolding

### LINKED ISSUE #46: Transaction scaffolding
child of #45

Basic struct & traits for Tari Transactions

### PR DESCRIPTION
**Pull request by CjS77**
## Description
Adds empty structs and traits for Tari/MW transactions

## Motivation and Context
Fixes #46 

## How Has This Been Tested?
`cargo test` passes locally

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/core/src/lib.rs [hunk 1]
```diff
@@ -20,6 +20,11 @@
 // WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 // USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+#[macro_use]
+extern crate bitflags;
```
[the remark this answers is not in the document]
Comment (CjS77) [reply base_layer/core/src/lib.rs]: Supposedly not, but I couldn't get the macro import working any other way :(

File: base_layer/core/src/transaction.rs [hunk 2]
```diff
@@ -19,11 +19,75 @@
...
+        const COINBASE_OUTPUT = 0b00000001;
+    }
+}
 
-pub struct TransactionInput;
+/// A transaction input.
+///
+/// Primarily a reference to an output being spent by the transaction.
+#[derive(Debug, Clone)]
+pub struct TransactionInput {
```
[the remark this answers is not in the document]
Comment (CjS77) [reply base_layer/core/src/transaction.rs]: In the `derive` clause?

[the remark this answers is not in the document]
Comment (philipr-za) [reply base_layer/core/src/transaction.rs]: The output below has the `Copy` so should probably maintain consistency

[the remark this answers is not in the document]
Comment (CjS77) [reply base_layer/core/src/transaction.rs]: Gotcha. Will address in followup PR. It's possible that we'll need custom Copy implementations.

**Pull request merged**